### PR TITLE
[SIG-2447] Refactor IncidentDetail Detail components

### PR DESCRIPTION
**Pull request by janjaap**
This PR contains changes to the `IncidentDetail` inner `Detail` components and its subsequent inner components.

Most notable changes:
- Throughout the inner components, the HTML structure is restructured so that the incident detail data is a single definition list, instead of multiple nested lists
- Components have been refactored into functional components
- `.scss` Files have been removed in favor of styled components

### CODE REVIEW

File: src/signals/incident-management/containers/IncidentDetail/components/Detail/components/ExtraProperties/index.js [hunk 1]
```diff
@@ -1,12 +1,10 @@
-import React from 'react';
+import React, { Fragment } from 'react';
 import isObject from 'lodash.isobject';
 import isArray from 'lodash.isarray';
 import isBoolean from 'lodash.isboolean';
 
 import { extraPropertiesType } from 'shared/types';
 
-import './style.scss';
```
Comment (aditudorache) [inline src/signals/incident-management/containers/IncidentDetail/components/Detail/components/ExtraProperties/index.js:8]: Can we remove the file as well?

File: src/signals/incident-management/containers/IncidentDetail/components/Detail/components/Location/index.js [hunk 2]
```diff
@@ -1,67 +1,78 @@
...
+      />
+
+      <MapTrigger onClick={onShowLocation} data-testid="location-button-show">
+        <MapDetail value={location} hideAttribution hideZoomControls useSmallMarker zoom="15" />
+      </MapTrigger>
+
+      {location.address_text ? (
+        <Address>
+          <div data-testid="location-value-address-stadsdeel">
+            Stadsdeel: {getListValueByKey(stadsdeelList, location.stadsdeel)}
```
Comment (aditudorache) [inline src/signals/incident-management/containers/IncidentDetail/components/Detail/components/Location/index.js:56]: Are we sure that the location properties exist?

File: src/signals/incident-management/containers/IncidentDetail/components/Detail/components/Location/index.js [hunk 3]
```diff
@@ -1,67 +1,78 @@
...
+          </div>
+          <div data-testid="location-value-address-street">
+            {location.address.openbare_ruimte} {location.address.huisnummer}
```
Comment (aditudorache) [inline src/signals/incident-management/containers/IncidentDetail/components/Detail/components/Location/index.js:59]: Are we sure that the location properties exist?

File: src/signals/incident-management/containers/IncidentDetail/components/Detail/components/Location/index.js [hunk 4]
```diff
@@ -1,67 +1,78 @@
...
+        </Address>
+      ) : (
+        <span data-testid="location-value-pinned">Locatie is gepind op de kaart</span>
+      )}
+    </Description>
+  </Fragment>
+);
 
 Location.propTypes = {
   incident: incidentType.isRequired,
```
Comment (aditudorache) [inline src/signals/incident-management/containers/IncidentDetail/components/Detail/components/Location/index.js:75]: Can't find the `location` definition in the `incidentType`

File: src/signals/incident-management/containers/IncidentDetail/components/Detail/index.js [hunk 5]
```diff
@@ -1,68 +1,90 @@
...
-          <dt className="detail__definition" data-testid="detail-email-definition">E-mail melder</dt>
-          <dd className="detail__value" data-testid="detail-email-value">{incident.reporter.email}</dd>
-
-          <dt className="detail__definition" data-testid="detail-phone-definition">Telefoon melder</dt>
-          <dd className="detail__value" data-testid="detail-phone-value">{incident.reporter.phone}</dd>
-        </dl>
-      </article>
-    );
+const DemiParagraph = styled(Paragraph)`
+  font-family: Avenir Next LT W01 Demi;
```
Comment (aditudorache) [inline src/signals/incident-management/containers/IncidentDetail/components/Detail/index.js:15]: I think we can enforce the use of this font automatically just by using `font-weight: 700`

File: src/signals/incident-management/containers/IncidentDetail/components/Detail/index.js [hunk 6]
```diff
@@ -1,68 +1,90 @@
...
+  margin: ${themeSpacing(4)} 0;
+`;
+
+const DefinitionList = styled.dl`
+  margin: 0;
+  display: grid;
```
Comment (aditudorache) [inline src/signals/incident-management/containers/IncidentDetail/components/Detail/index.js:21]: Did we drop IE11 support? I thought the grid is not good supported on IE11.

File: src/signals/incident-management/containers/IncidentDetail/components/Detail/components/ExtraProperties/index.js (hunk 1 above)
Comment (janjaap) [reply src/signals/incident-management/containers/IncidentDetail/components/Detail/components/ExtraProperties/index.js:8]: Done

File: src/signals/incident-management/containers/IncidentDetail/components/Detail/components/Location/index.js (hunk 3 above)
Comment (janjaap) [reply src/signals/incident-management/containers/IncidentDetail/components/Detail/components/Location/index.js]: Good one; I've made sure that address parts are conditionally rendered

File: src/signals/incident-management/containers/IncidentDetail/components/Detail/components/Location/index.js (hunk 4 above)
Comment (janjaap) [reply src/signals/incident-management/containers/IncidentDetail/components/Detail/components/Location/index.js:81]: Sharp as a knife; added it.

File: src/signals/incident-management/containers/IncidentDetail/components/Detail/index.js (hunk 5 above)
Comment (janjaap) [reply src/signals/incident-management/containers/IncidentDetail/components/Detail/index.js:15]: How would we do that?

File: src/signals/incident-management/containers/IncidentDetail/components/Detail/index.js (hunk 6 above)
Comment (janjaap) [reply src/signals/incident-management/containers/IncidentDetail/components/Detail/index.js:21]: No, we didn't (officially). The use of CSS grid layout gracefully degrades, so it still looks okay.

File: src/signals/incident-management/containers/IncidentDetail/components/Detail/index.js (hunk 5 above)
Comment (aditudorache) [reply src/signals/incident-management/containers/IncidentDetail/components/Detail/index.js:15]: This works if you use asc-assets fonts:
- in global.css (and overall where an other name is used): 
```
body {
  font-family: 'Avenir Next', 'Helvetica Neue', Helvetica, Arial, sans-serif;
}
```
- in webpack
```
    new CopyWebpackPlugin([
      {
        from: './node_modules/@datapunt/asc-assets/static/fonts',
        to: 'fonts',
      },
    ]),
```
- in index.html
```
    <!-- This is the tracking code for the fastfonts.net typography -->
    <script src="/fonts/mtiFontTrackingCode.min.js"></script>
    <link href="/fonts/fonts.css" rel="stylesheet" />

```
- in the rest of the application use `font-weight: 700` for demi and , `font-weight: 500`  for medium.

see './node_modules/@datapunt/asc-assets/static/fonts' for the font face options. Only the design system fonts are defined there

Comment (janjaap) [reply src/signals/incident-management/containers/IncidentDetail/components/Detail/index.js:15]: Ah, maybe for another pull request

**Pull request merged**